src/widgets/fields/baseField.lua

```diff
@@ -0,0 +1,163 @@
+local Runtime = require(script.Parent.Parent.Parent.Runtime)
+local Style = require(script.Parent.Parent.Parent.Style)
+local create = require(script.Parent.Parent.Parent.create)
+local automaticSize = require(script.Parent.Parent.Parent.automaticSize)
+local Field = require(script.Parent.Parent.Parent.Field)
+
+local DEPTH_PADDING = 20
+
+local fieldValueDisplay = Runtime.widget(function(value, options, globalFieldOptions)
+	local fieldValueType = options.fieldValueType or Field.FieldValueType.TEXTBOX
+
+	if fieldValueType == Field.FieldValueType.TEXTBOX then
+		local refs = Runtime.useInstance(function(ref)
+			local style = Style.get()
+			return create("TextBox", {
+				[ref] = "valueText",
+				Size = UDim2.fromScale(1, 1),
+				TextXAlignment = Enum.TextXAlignment.Left,
+				TextColor3 = style.textColor,
+				BackgroundTransparency = 1,
+				Text = value,
+				FocusLost = function(enterPressed)
+					if enterPressed then
+						options.update(ref.valueText.Text)
+					end
+				end,
+			})
+		end)
+
+		if not refs.valueText:IsFocused() then
+			refs.valueText.Text = value
+		end
+	elseif fieldValueType == Field.FieldValueType.TEXTLABEL then
+		Runtime.useInstance(function(ref)
+			local style = Style.get()
+			return create("TextLabel", {
+				Size = UDim2.fromScale(1, 1),
+				TextXAlignment = Enum.TextXAlignment.Left,
+				TextColor3 = style.textColor,
+				BackgroundTransparency = 1,
+				Text = value,
+			})
+		end)
+	end
+end)
+
+local fieldRow = Runtime.widget(function(key, value, options, globalFieldOptions)
```

Comment:
What is the `value` parameter used for?

Reply:
The `value` parameter represents the value of the field. However, in this context it's being passed into `	fieldValueDisplay(value, options, globalFieldOptions)` so it has no purpose in fieldRow technically.